src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobFileStore.cs

```diff
@@ -183,6 +184,27 @@ public async Task CopyFileAsync(VersionedInstanceIdentifier versionedInstanceIde
         }
     }
 
+    /// <inheritdoc/>
+    public async Task DeleteOldFileIfExistsAsync(
+        VersionedInstanceIdentifier versionedInstanceIdentifier,
+        CancellationToken cancellationToken)
+    {
+        EnsureArg.IsNotNull(versionedInstanceIdentifier, nameof(versionedInstanceIdentifier));
+
+        var blobClient = GetInstanceBlockBlobClient(versionedInstanceIdentifier, BlobMigrationFormatType.Old);
+        var newBlobClient = GetInstanceBlockBlobClient(versionedInstanceIdentifier, BlobMigrationFormatType.New);
+
+        // Delete the old file only if the new file exists
+        if (await newBlobClient.ExistsAsync(cancellationToken))
+        {
+            await ExecuteAsync(() => blobClient.DeleteIfExistsAsync(DeleteSnapshotsOption.IncludeSnapshots, conditions: null, cancellationToken));
```

Comment:
Is it necessary to call `ExistsAsync` if we're going to call `DeleteIfExistsAsync`? Presumably `DeleteIfExistsAsync` returns a result based on whether the blob exists, and we could use that for the log below?

Reply:
Yes, we need to make sure new file exists before deleting the old one.

Reply:
Oh, I misread this